blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java

```diff
@@ -242,9 +242,9 @@ protected void handleChunkDone(@Nonnull MemoryFlowChunk chunk) {
             if (parallelNestedStages && chunk.getFirstNode().isActive()) {
                 status = new NodeRunStatus(chunk.getFirstNode());
             } else {
-                FlowNode nodeBefore = chunk.getNodeBefore(), lastNode = chunk.getLastNode(), nodeAfter = chunk.getNodeAfter();
+                FlowNode nodeBefore = chunk.getNodeBefore(), firstNode = chunk.getFirstNode(), lastNode = chunk.getLastNode(), nodeAfter = chunk.getNodeAfter();
                 status = new NodeRunStatus(
-                    StatusAndTiming.computeChunkStatus2(run, nodeBefore, firstExecuted, lastNode, nodeAfter));
+                    StatusAndTiming.computeChunkStatus2(run, nodeBefore, firstNode, lastNode, nodeAfter));
```

Comment:
I thought this change wasn't strictly necessary, but it turns out that it is, and it fixes things like [JENKINS-57579](https://issues.jenkins-ci.org/browse/JENKINS-57579). I can't quite tell what `firstExecuted` is expected to be, and it seems like it will vary based on context, but `firstNode` is _always_ (even for in-progress builds), going to be the start of a stage step block or parallel branch block, and that is what we expect over in `StatusAndTiming.computeChunkStatus2` anyway.

My guess as to the problem is that in some cases `firstExecuted` ends up being identical to or later than `lastNode` in the graph, and so setting up the graph traversal [here in StatusAndTiming.findWarningsBetween](https://github.com/jenkinsci/pipeline-graph-analysis-plugin/blob/a34a23633dc8ebd4c74012498335653b69b45206/src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java#L326) fails and the method returns null.